spirv_reflect.c

```diff
@@ -3513,32 +3531,7 @@ static SpvReflectResult ParseExecutionModes(
         }
         break;
 
-        case SpvExecutionModeOutputPoints:
-        case SpvExecutionModeOutputLineStrip:
-        case SpvExecutionModeOutputTriangleStrip:
-        case SpvExecutionModeVecTypeHint:
-        case SpvExecutionModeContractionOff:
-        case SpvExecutionModeInitializer:
-        case SpvExecutionModeFinalizer:
-        case SpvExecutionModeSubgroupSize:
-        case SpvExecutionModeSubgroupsPerWorkgroup:
-        case SpvExecutionModeSubgroupsPerWorkgroupId:
-        case SpvExecutionModeLocalSizeId:
-        case SpvExecutionModeLocalSizeHintId:
-        case SpvExecutionModePostDepthCoverage:
-        case SpvExecutionModeDenormPreserve:
-        case SpvExecutionModeDenormFlushToZero:
-        case SpvExecutionModeSignedZeroInfNanPreserve:
-        case SpvExecutionModeRoundingModeRTE:
-        case SpvExecutionModeRoundingModeRTZ:
-        case SpvExecutionModeStencilRefReplacingEXT:
-        case SpvExecutionModeOutputLinesNV:
-        case SpvExecutionModeOutputPrimitivesNV:
-        case SpvExecutionModeOutputTrianglesNV:
-        case SpvExecutionModePixelInterlockOrderedEXT:
-        case SpvExecutionModePixelInterlockUnorderedEXT:
-        case SpvExecutionModeSampleInterlockOrderedEXT:
-        case SpvExecutionModeSampleInterlockUnorderedEXT:
+        default:
```

Comment:
For this change, what happens if we encounter an execution mode that we haven't seen before? 

I see the addition of `SPV_REFLECT_EXECUTION_MODE_UNKNOWN` but not seeing it used.

Reply:
ya, can remove `SPV_REFLECT_EXECUTION_MODE_UNKNOWN` that was going to be for the `WorkgroupSize` support I decided to defer

> what happens if we encounter an execution mode that we haven't seen before?

Well before when we got something like `SpvExecutionModeOutputPoints` we just did nothing anyways, we track the values in the `spirv.h` so if they are using a valid execution mode it just works... if they aren't, well they somehow got pass all the other SPIR-V tools the would have blown up first

Reply:
> ya, can remove `SPV_REFLECT_EXECUTION_MODE_UNKNOWN` that was going to be for the `WorkgroupSize` support I decided to defer

Can leave it in if you plan to use it later. Not to fussed about it.

> > what happens if we encounter an execution mode that we haven't seen before?
> 
> Well before when we got something like `SpvExecutionModeOutputPoints` we just did nothing anyways, we track the values in the `spirv.h` so if they are using a valid execution mode it just works... if they aren't, well they somehow got pass all the other SPIR-V tools the would have blown up first

Got it. I guess we'd get a test would blow up or we'll get a bug report if it's too egregious.

Reply:
> Can leave it in if you plan to use it later. Not to fussed about it.

Might be a while, better to just added when needed (also removed it)